Sources/Base/Types.swift

```diff
@@ -36,7 +42,15 @@ enum DataType: CustomStringConvertible, Hashable {
     return .array(field: field, length: nil)
   }
   static let string = DataType.custom(name: "String")
-  
+  static var freshTypeVariable : DataType {
+    defer { DataType.typeVariablePool += 1 }
+    return .typeVariable(name: "T\(DataType.typeVariablePool)")
+  }
+  static var freshMetaVariable : DataType {
```

Comment:
Meta variables are spawned and solved during unification, type variables are spawned and solved by CSGen.  You can merge both definitions, but I kept them separate because of the paper I was reading while writing this.